src/com/selftest/stack/StackImpl.java

```diff
@@ -0,0 +1,59 @@
+
+package com.selftest.stack;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class StackImpl<T> implements Stack<T> {
+
+	private T[] stack;
+	private int size;
+
+	@SafeVarargs
+	public StackImpl( int size, T ... values ) {
+		stack = Arrays.copyOf( values, size );
+		size = 0;

```

Comment:
Hmm, I would imagine that would be a surprise feature.  I forgot to update the index (size)...but I am sure I _would_ have picked that up in unit testing :)